slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -46,64 +88,140 @@ SLAMServiceActionMode SLAMServiceImpl::GetActionMode() {
 
 void SLAMServiceImpl::OverwriteMapBuilderParameters() {
     SLAMServiceActionMode slam_action_mode = GetActionMode();
-
-    auto mutable_pose_graph_options =
-        mapBuilder.map_builder_options_.mutable_pose_graph_options();
-    mutable_pose_graph_options->set_optimize_every_n_nodes(
-        optimize_every_n_nodes);
-
-    auto mutable_trajectory_builder_2d_options =
-        mapBuilder.trajectory_builder_options_
-            .mutable_trajectory_builder_2d_options();
-    mutable_trajectory_builder_2d_options->mutable_submaps_options()
-        ->set_num_range_data(num_range_data);
-    mutable_trajectory_builder_2d_options->set_missing_data_ray_length(
-        missing_data_ray_length);
-    mutable_trajectory_builder_2d_options->set_max_range(max_range);
-    mutable_trajectory_builder_2d_options->set_min_range(min_range);
-    if (slam_action_mode == SLAMServiceActionMode::LOCALIZING) {
-        mapBuilder.trajectory_builder_options_
-            .mutable_pure_localization_trimmer()
-            ->set_max_submaps_to_keep(max_submaps_to_keep);
-    }
-
-    auto mutable_overlapping_submaps_trimmer_2d =
-        mutable_pose_graph_options->mutable_overlapping_submaps_trimmer_2d();
-    if (slam_action_mode == SLAMServiceActionMode::UPDATING) {
-        mutable_overlapping_submaps_trimmer_2d->set_fresh_submaps_count(
-            fresh_submaps_count);
-        mutable_overlapping_submaps_trimmer_2d->set_min_covered_area(
-            min_covered_area);
-        mutable_overlapping_submaps_trimmer_2d->set_min_added_submaps_count(
-            min_added_submaps_count);
+    {
+        std::lock_guard<std::mutex> lk(map_builder_mutex);
+        map_builder.OverwriteOptimizeEveryNNodes(optimize_every_n_nodes);
+        map_builder.OverwriteNumRangeData(num_range_data);
+        map_builder.OverwriteMissingDataRayLength(missing_data_ray_length);
+        map_builder.OverwriteMaxRange(max_range);
+        map_builder.OverwriteMinRange(min_range);
+        if (slam_action_mode == SLAMServiceActionMode::LOCALIZING) {
+            map_builder.OverwriteMaxSubmapsToKeep(max_submaps_to_keep);
+        }
+        if (slam_action_mode == SLAMServiceActionMode::UPDATING) {
+            map_builder.OverwriteFreshSubmapsCount(fresh_submaps_count);
+            map_builder.OverwriteMinCoveredArea(min_covered_area);
+            map_builder.OverwriteMinAddedSubmapsCount(min_added_submaps_count);
+        }
+        map_builder.OverwriteOccupiedSpaceWeight(occupied_space_weight);
+        map_builder.OverwriteTranslationWeight(translation_weight);
+        map_builder.OverwriteRotationWeight(rotation_weight);
```

Comment:
Moved overwriting internal `MapBuilder` parameters the responsibility of `MapBuilder` itself, rather than letting the slam service do it. Found it also easier to read.